[catch2]当全局工具链指定为`clang-cl`，链接依赖`catch2`程序时，连接参数中会丢失`-subsystem:console`标志

### Xmake 版本

v2.9.6

### 操作系统版本和架构

Windows10 22H2

### 描述问题

当全局工具链指定为`clang-cl`，链接依赖`catch2`程序时，连接参数中会丢失`-subsystem:console`标志，导致链接时无法找到Catch2Main.lib中的`main`函数

### 期待的结果

不丢失参数

### 工程配置

```lua
set_runtimes("MD")

add_requires("catch2")
set_toolchains("clang-cl")

target("main")
    set_kind("binary")
    
    add_files("./*.cpp")
    add_packages("catch2")

```

### 附加信息和错误日志

```console
checking for D:\scoop\apps\llvm\current\bin\clang-cl.exe ... ok
checking for flags (cl_external_includedir) ... ok
[ 75%]: linking.release main.exe
"D:\\Microsoft Visual Studio\\2022\\Community\\VC\\Tools\\MSVC\\14.43.34604\\bin\\HostX64\\x64\\link.exe" -nologo -dynamicbase -nxcompat -machine:x64 -libpath:F:\xmake\.xmake\packages\c\catch2\v3.7.1\475769e922364ab3992a0838c26b438e\lib Catch2Main.lib Catch2.lib -out:build\windows\x64\release\main.exe build\.objs\main\windows\x64\release\main.cpp.obj
error: LINK : fatal error LNK1561: 必须定义入口点
```

--- a/packages/c/catch2/xmake.lua
+++ b/packages/c/catch2/xmake.lua
@@ -69,7 +69,7 @@ package("catch2")
         end
         component:add("links", link)
         if package:is_plat("windows") and package:version():le("3.0") then
-            if package:has_tool("cxx", "cl") then
+            if package:has_tool("cxx", "cl", "clang-cl") then
                 component:add("ldflags", "-subsystem:console")
             elseif package:has_tool("cxx", "clang", "clangxx") then
                 component:add("ldflags", "-Wl,/subsystem:console")

--- a/packages/c/catch2/xmake.lua
+++ b/packages/c/catch2/xmake.lua
@@ -81,7 +81,7 @@ package("catch2")
         if package:version():ge("3.0") then
             if package:is_plat("windows") then
                 local main_component = package:component("main")
-                if package:has_tool("cxx", "cl") then
+                if package:has_tool("cxx", "cl", "clang-cl") then
                     main_component:add("ldflags", "-subsystem:console")
                 elseif package:has_tool("cxx", "clang", "clangxx") then
                     main_component:add("ldflags", "-Wl,/subsystem:console")

